managed/services/grafana/auth_server.go

```diff
@@ -280,35 +273,41 @@ func (s *AuthServer) returnError(rw http.ResponseWriter, msg map[string]any, l *
 
 // maybeAddVMProxyFilters adds extra filters to requests proxied through VMProxy.
 // In case the request is not proxied through VMProxy, this is a no-op.
-func (s *AuthServer) maybeAddVMProxyFilters(ctx context.Context, rw http.ResponseWriter, req *http.Request, userID int, l *logrus.Entry) error {
+func (s *AuthServer) maybeAddVMProxyFilters(
+	ctx context.Context, rw http.ResponseWriter, req *http.Request,
+	userID int, teamIDs []int, l *logrus.Entry,
+) error {
 	if !s.shallAddVMProxyFilters(req) {
 		return nil
 	}
 
-	if userID == 0 {
+	var authUser *authUser
+	if userID == 0 || teamIDs == nil {
 		l.Debugf("Getting authenticated user info")
-		authUser, err := s.getAuthUser(ctx, req, l)
+		var err *authError
+		authUser, err = s.getAuthUser(ctx, req, l, true)
 		if err != nil {
-			return ErrCannotGetUserID
+			return errors.Errorf("%w\n%w", ErrCannotGetUserID, err)
 		}
 
 		if authUser == nil {
-			return fmt.Errorf("%w: user is empty", ErrCannotGetUserID)
+			return errors.Errorf("%w: user is empty", ErrCannotGetUserID)
 		}
 
 		userID = authUser.userID
+		teamIDs = authUser.teamIDs
 	}
 
-	if userID <= 0 {
+	if userID <= 0 || teamIDs == nil {
 		return ErrInvalidUserID
```

Comment:
I am not sure, but maybe in case of no team we should return different error message.